File: Baekjoon_Online_Judge/NO_11256/11256.cpp
#include <iostream>
#include <algorithm>
using namespace std;
/*
 ---------------------------- 
 | Created by happiness96   | 
 | Year 2020                | 
 | Month 07                 | 
 | Day 28                   | 
 | 11256 사탕                | 
 ---------------------------- 
*/

int main(){
    int T;
    
    cin >> T;

    for(int t = 0; t < T; t ++){
        int J, N;

        cin >> J >> N;
        
        int box[N];

        for(int i = 0; i < N; i ++){
            int R, C;

            cin >> R >> C;
            box[i] = R * C;
        }

        sort(box, box + N);

        for(int i = N - 1; i >= 0; i --){
            J -= box[i];
            
            if(J <= 0){
                cout << N - i << endl;
                break;
            }
        }

    }

    return 0;
}

File: Baekjoon_Online_Judge/NO_11256/11256.py
# -*- encoding: utf-8 -*-
import sys
r_input = sys.stdin.readline

# ---------------------------- #
# | Created by happiness96   | #
# | Year 2020                | #
# | Month 07                 | #
# | Day 28                   | #
# | 11256 사탕                | #
# ---------------------------- #


def run():
    T = int(r_input())

    for _ in range(T):
        J, N = map(int, r_input().split())

        save = []

        for _ in range(N):
            R, C = map(int, r_input().split())
            save.append(R * C)
        
        save.sort()

        for ind, val in enumerate(save[::-1]):
            J -= val

            if J <= 0:
                print(ind + 1)
                break


if __name__ == "__main__":
    run()
